lib/src/source_visitor.dart

```diff
@@ -108,12 +108,45 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   visitAdjacentStrings(AdjacentStrings node) {
     builder.startSpan();
-    builder.startRule();
-    visitNodes(node.strings, between: splitOrNewline);
+    var rule = new Rule();
+    builder.startRule(rule);
+
+    var chunk = zeroSplit();
```

Comment:
Oof, that's hard. You don't actually always want to ensure there's a split before the first string. Consider:

```dart
{
  "long adjacent" "long adjacent";
}
```

That should not get formatted to:

```dart
{
  // <-- split added here.
  "long adjacent"
  "long adjacent";
}
```

Likewise:

```dart
var x = foo ? "short" : "long adjacent" "long adjacent";
```

Should maybe not become:

```dart
var x = foo
    ? "short"
    :
        "long adjacent"
        "long adjacent";
```

Or maybe it should? It may be that the best way to handle this is to have ArgumentListVisitor specifically handle arguments that are adjacent strings.